sdk/openai/Azure.AI.OpenAI/src/Custom/OpenAIClient.cs

```diff
@@ -42,5 +45,73 @@ public virtual Response<Completions> GetCompletions(string deploymentId, string
             completionsOptions.Prompt.Add(prompt);
             return GetCompletions(deploymentId, completionsOptions, cancellationToken);
         }
+
+        public virtual Response<StreamingCompletions> GetStreamingCompletions(
+            string deploymentId,
+            CompletionsOptions completionsOptions,
+            CancellationToken cancellationToken = default)
+        {
+            Argument.AssertNotNullOrEmpty(deploymentId, nameof(deploymentId));
+            Argument.AssertNotNull(completionsOptions, nameof(completionsOptions));
+
+            using DiagnosticScope scope = ClientDiagnostics.CreateScope(
+                "OpenAIClient.GetStreamingCompletions");
+            scope.Start();
+
+            RequestContext context = FromCancellationToken(cancellationToken);
+
+            RequestContent nonStreamingContent = completionsOptions.ToRequestContent();
+            RequestContent streamingContent = GetStreamingEnabledRequestContent(nonStreamingContent);
```

Comment:
it's fine for the preview, but we should fix it at some point, i.e. we should generate the right content the first time (line above) as opposed to rewriting it.

Reply:
Agreed on both points -- I'd like to integrate the in-progress proper approach ASAP (maybe even before this PR is done) but won't hesitate to use this frowny-face-inducing workaround if it's the holdout.